Automattic/wp-calypso
# Manage Purchase: use site level query component in site level context

We're currently using the user level purchase query component in ManagePurchase in both the site-level and account-level contexts, which for users with a lot of sites is extremely slow.

#### Changes proposed in this Pull Request

* Conditionally use the site level query component for loading purchases. The first 4 commits handle this; the rest are refactoring `PurchaseMeta` to a functional component.

#### Testing instructions

We need to test for two things: that performance improves in the site-level context, and that the behavior is preserved in the account-level context.

Site-level:
- Navigate to `/purchases/subscriptions/:site_url`.
- Wait for your list of purchases to load.
- Select a purchase to navigate to the Purchase Settings screen, and check that this screen loads correctly.
- In the network tab, check that you do not see an api request to `me/purchases`.
- To see the performance improvement, you'll need to test as a user with lots of purchases. For example, my test user has ~60 purchases, and the request to `me/purchases` takes about a minute, but `sites/:site_id/purchases` is about 500ms.

Account-level:
- Navigate to `/me/purchases`.
- Wait for your list of purchases to load (should take longer if your user has a lot of purchases).
- Select a purchase to navigate to the Purchase Settings screen, and check that the Purchase Settings screen loads correctly.
